### DISCUSSION TITLE
[4.0] Content Security Policy Reports - questions

### DISCUSSION THREAD

**Discussion by brianteeman** [joomla/joomla-cms#31641] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/1296369/101993348-bb0cfa80-3cb1-11eb-8093-00b222e8b13e.png)

1. What is the purpose of the URL column?
2. Would it be better to say enabled/disabled instead of published/unpublished?

@zero-24

**Comment by zero-24** [comment]
> What is the purpose of the URL column?

The original site what that report is comming from.

> Would it be better to say enabled/disabled instead of published/unpublished?

Whatever is best. Enabled / Disabled should work too IIRC published/unpublished is just the default :)

**Comment by brianteeman** [comment]
>> What is the purpose of the URL column?

> The original site what that report is comming from.

Wouldn't that be the domain then and not the URL ?

>>Would it be better to say enabled/disabled instead of published/unpublished?

> Whatever is best. Enabled / Disabled should work too IIRC published/unpublished is just the default :)

For me it is like plugins and should be enabled etc. I will prepare a PR for that

**Comment by zero-24** [comment]
> Wouldn't that be the domain then and not the URL ?

One site only has (usually) only one domain right? When the report comes from example.org/joomla/10-joomla than this is the URL? Or I'm missing something?

**Comment by brianteeman** [comment]
So then there is a bug as all my entries in the column are identical as shown in the screenshot

**Comment by brianteeman** [comment]
PR for the enabled instead of publish #31643

**Comment by zero-24** [comment]
> So then there is a bug as all my entries in the column are identical as shown in the screenshot

Well do you have other sub pages where specific stuff is loaded that is not loaded on the main page? For example a youtube page?

**Comment by zero-24** [comment]
> PR for the enabled instead of publish #31643

Thanks.

**Comment by brianteeman** [comment]
The site is installed at http://localhost/joomla-cms

Below is the results after having detect mode on for site and admin and then visiting multiple pages on both. As you can see they all have the identical data in the URI column - I was expecting the full link to the page so that I am able to see what is causing the report.

![image](https://user-images.githubusercontent.com/1296369/102262468-0c63f680-3f0b-11eb-93b6-5876242a7df1.png)

**Comment by zero-24** [comment]
> I was expecting the full link to the page so that I am able to see what is causing the report.

Yes that is or better was the intended behavior seems to be a bug to me. Also seems that [I actually implemented that issue](https://github.com/joomla/joomla-cms/commit/f8bf0ba08bafa1659dd7aaf2c196ed9069299319#diff-8d5dc402732cb296a1e6e4d21cbc9125b4943580c4c6480e304a068d9c41027cR127) in there.

In the end that line should be something like:

```
$report->blocked_uri = $parsedBlockedUri['scheme'] . '://' . $parsedBlockedUri['host'] . '/' .  $parsedBlockedUri['path'] . '/' . $parsedBlockedUri['query'];
```

Maybe  we even better throw that into the Uri class?

**Comment by richard67** [comment]
Should we move this to discussions?

**Reply by zero-24** [reply]
Why should we its a bug as discussed with Brian above?

**Reply by richard67** [reply]
The description was reading like a question to me so I had asked if we shall move it. But it wasn't me who finally moved it.

**Comment by brianteeman** [comment]
No - its a bug

**Comment by brianteeman** [comment]
This is a BUG not a discussion Please move it back!!!

**Reply by richard67** [reply]
It wasn't me who moved it. I had asked before if we shall move it or not because to me the issue description was reading like a question and not like a bug report. After that it was moved by Robert.

Unfortunately there is no way (yet) to convert a discussion back to an issue, but if it turns out that a discussion is an issue /bug), then of course a new issue should be created for that.

**Reply by PhilETaylor** [reply]
There also appears to be no way to mark a discussion as closed... or deleted...

**Comment by brianteeman** [comment]
accidents happen - please move it back!!

**Reply by PhilETaylor** [reply]
You can't. That is not a feature of discussions. You cannot convert a discussion to an issue.

Its also not an accident if its happened multiple times.

**Comment by brianteeman** [comment]
Created a new BUG report as an issue https://github.com/joomla/joomla-cms/issues/31709

**Comment by brianteeman** [comment, marked as answer]
CLOSED

**Comment by richard67** [comment]
I've moved it to "Q & A" so it can be marked as solved.

Thanks @brianteeman for having created the new issue #31709 , and sorry for the inconvenience.
